src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -29,48 +27,267 @@ export const CollectorSignals: GraphQLFieldConfig<any, ResolverContext> = {
       bidCount: {
         type: GraphQLInt,
         description: "Bid count on lots open for bidding",
+        resolve: async (artwork, {}, ctx) => {
+          const isInSale = artwork.sale_ids?.length > 0
+
+          if (
+            !checkFeatureFlag("emerald_signals-auction-improvements", ctx) ||
+            !isInSale
+          ) {
+            return null
+          }
+
+          const activeLotData = await getActiveSaleArtwork(
+            {
+              artworkId: artwork.id,
+              saleIds: artwork.sale_ids,
+            },
+            ctx
+          )
+
+          if (activeLotData) {
+            return activeLotData.saleArtwork.bidder_positions_count
+          }
+
+          return null
+        },
       },
       lotWatcherCount: {
         type: GraphQLInt,
         description: "Lot watcher count on lots open for bidding",
+        resolve: async (artwork, {}, ctx) => {
+          const isInSale = artwork.sale_ids?.length > 0
+
+          if (
+            !checkFeatureFlag("emerald_signals-auction-improvements", ctx) ||
+            !isInSale
+          ) {
+            return null
+          }
+
+          const activeLotData = await getActiveSaleArtwork(
+            {
+              artworkId: artwork.id,
+              saleIds: artwork.sale_ids,
+            },
+            ctx
+          )
+
+          if (activeLotData) {
+            return artwork.recent_saves_count
+          }
+
+          return null
+        },
       },
       registrationEndsAt: {
         type: GraphQLString,
         description: "Pending auction registration end time",
+        resolve: async (artwork, {}, ctx) => {
+          const isInSale = artwork.sale_ids?.length > 0
+
+          if (
+            !checkFeatureFlag("emerald_signals-auction-improvements", ctx) ||
+            !isInSale
+          ) {
+            return null
+          }
+
+          const activeLotData = await getActiveSaleArtwork(
+            {
+              artworkId: artwork.id,
+              saleIds: artwork.sale_ids,
+            },
+            ctx
+          )
+
+          if (activeLotData) {
+            const registrationEndAtDate =
+              activeLotData.sale.registration_ends_at &&
+              new Date(activeLotData.sale.registration_ends_at)
+
+            if (registrationEndAtDate && registrationEndAtDate > NOW) {
+              return registrationEndAtDate.toISOString()
+            }
+          }
+
+          return null
+        },
       },
       lotClosesAt: {
         type: GraphQLString,
```

Comment:
If this returns a timestamp, should be using our `date` field factory - https://github.com/artsy/metaphysics/blob/d1419db333d2959bd8b81e362dc247013f92bf9a/src/schema/v2/fields/date.ts